controller.go

```diff
@@ -181,6 +258,47 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 	return network, nil
 }
 
+func (c *controller) newNetworkFromStore(n *network) {
+	c.Lock()
+	n.ctrlr = c
+	n.driver = c.drivers[n.networkType]
+	c.networks[n.id] = n
+	c.Unlock()
+	// TODO : Populate n.endpoints back from endpoint dbstore
+}
+
+func (c *controller) addNetworkToStore(n *network) error {
+	if isReservedNetwork(n.Name()) {
+		return nil
+	}
+	if c.store == nil {
+		return ErrInvalidDatastore
+	}
+	return c.store.PutObjectAtomic(n)
+}
+
+func (c *controller) watchNewNetworks() {
+	c.store.KVStore().WatchRange(datastore.Key("network"), "", 0, func(kvi []store.KVEntry) {
```

Comment:
Take a reference to `c.store` under a lock and use that to invoke `store.KVStore`